test/tests/functional/pbs_job_array.py

```diff
@@ -982,3 +987,32 @@ def test_max_run_subjobs_eligible_time(self):
         self.server.expect(JOB, {'job_state=R': 3}, extend='t')
         self.server.expect(JOB, {'accrue_type': accrue['eligible']},
                            id=jid_arr)
+
+    def test_max_run_subjobs_on_non_array(self):
+        """
+        Test that setting max_run_subjobs on non-array jobs is rejected.
+        """
+
+        qsub_cmd = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                                'bin', 'qsub')
+
+        cmd = [qsub_cmd, '-Wmax_run_subjobs=4', '--', '/bin/sleep 100']
+        rv = self.du.run_cmd(self.server.hostname, cmd=cmd)
+        self.assertNotEqual(rv['rc'], 0, 'qsub must fail')
+        self.assertEqual(rv['err'][0], "qsub: Not an array job")
+
+    def test_multiple_max_run_subjobs_values(self):
+        """
+        Test that setting max_run_subjobs more than once on an array
+        job is rejected.
+        """
+
+        qsub_cmd = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                                'bin', 'qsub')
+
+        cmd = [qsub_cmd, '-J1-4%2', '-Wmax_run_subjobs=4', '--',
+               '/bin/sleep 100']
+        rv = self.du.run_cmd(self.server.hostname, cmd=cmd)
+        self.assertNotEqual(rv['rc'], 0, 'qsub must fail')
+        msg = "qsub: multiple max_run_subjobs values found"
+        self.assertEqual(rv['err'][0], msg)
```

Comment:
You can do this with self.server.submit().  You do array_indices_submitted = '1-4%2'

Reply:
I think for this, it has to come through qsub because the error check is in qsub command itself.